File: packages/firestore/ios/Plugin/Classes/Options/UpdateDocumentOptions.swift
import Foundation
import Capacitor

@objc public class UpdateDocumentOptions: NSObject {
    private var reference: String
    private var data: [String: Any]
    private var databaseId: String?

    init(reference: String, data: JSObject, databaseId: String? = nil) {
        self.reference = reference
        self.data = FirebaseFirestoreHelper.createHashMapFromJSObject(data)
        self.databaseId = databaseId
    }

    func getReference() -> String {
        return reference
    }

    func getData() -> [String: Any] {
        return data
    }

    func getDatabaseId() -> String? {
        return databaseId
    }
}

File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/UpdateDocumentOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestoreHelper;
import java.util.Map;
import org.json.JSONException;

public class UpdateDocumentOptions {

    private String reference;
    private Map<String, Object> data;
    @Nullable
    private String databaseId;

    public UpdateDocumentOptions(String reference, JSObject data, @Nullable String databaseId) throws JSONException {
        this.reference = reference;
        this.data = FirebaseFirestoreHelper.createHashMapFromJSONObject(data);
        this.databaseId = databaseId;
    }

    public String getReference() {
        return reference;
    }

    public Map<String, Object> getData() {
        return data;
    }

    @Nullable
    public String getDatabaseId() {
        return databaseId;
    }
}
